src/filter.rs

```diff
@@ -234,6 +240,60 @@ pub fn emboss(mut src: &mut Image) -> RasterResult<()>{
     convolve(src, matrix, 1)
 }
 
+/// Apply sobel.
+///
+/// # Examples
+/// ```
+/// use raster::{filter, Orientation};
+///
+/// // Create image from file
+/// let mut image = raster::open("tests/in/sample.jpg").unwrap();
+/// filter::sobel(&mut image, Orientation::Horizontal).unwrap();
+/// raster::save(&image, "tests/out/test_filter_sobel.jpg").unwrap();
+/// ```
+///
+/// ### Before
+/// ![](https://kosinix.github.io/raster/in/sample.jpg)
+///
+/// ### After
+/// ![](https://kosinix.github.io/raster/out/test_filter_sobel.jpg)
+///
+pub fn sobel(mut src: &mut Image, mode: Orientation) -> RasterResult<()> {
+    grayscale(src).unwrap();
+    let matrix: [[i32; 3]; 3];
+
+    match mode {
```

Comment:
This is a minor gripe, but it's arguably more clear to directly set `matrix` to the result of this `match` expression; early returns allow for that.